libbeat/outputs/elasticsearch/client.go

```diff
@@ -263,9 +264,10 @@ func (client *Client) publishEvents(ctx context.Context, data []publisher.Event)
 	pubCount := len(data)
 	span.Context.SetLabel("events_published", pubCount)
 
+	timeSinceSend := time.Since(begin)
 	client.log.Debugf("PublishEvents: %d events have been published to elasticsearch in %v.",
 		pubCount,
-		time.Since(begin))
+		timeSinceSend)
```

Comment:
I would expect the `time.Now()` and `time.Since()` to be right before and right after the call to `client.conn.Bulk`.  And we should record the value even if there is an error, like StatusRequestEntityTooLarge.  And thinking on that error we should probably have a counter for it as well.

Reply:
The metrics observer already has a `Failed` counter, unless you're thinking of something else?

Reply:
we don't have a counter for how many times we hit `StatusRequestEntityTooLarge`, which would be good to have, so you know what to fix stop the failing events.

Reply:
The current observer mechanism seems like it's meant to be somewhat generic? Should we make a follow-up issue for tracking `StatusRequestEntityTooLarge` ?

Reply:
nevermind, the Publish function catches this https://github.com/elastic/beats/blob/5fab3e9b1a93305cc84f73504d31df0c796d7047/libbeat/outputs/elasticsearch/client.go#L196C4-L196C27